specification/containerservice/Fleet.Management/update/autoupgradeprofile.tsp

```diff
@@ -0,0 +1,108 @@
+import "@typespec/rest";
+import "@typespec/openapi";
+import "@typespec/versioning";
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-azure-resource-manager";
+import "./../helpers.tsp";
+import "./run.tsp";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+using Azure.ResourceManager;
+using Azure.ResourceManager.Foundations;
+using Azure.Core;
+using Azure.Core.Traits;
+using TypeSpec.OpenAPI;
+
+namespace Microsoft.ContainerService;
+
+@resource("autoUpgradeProfiles")
+@parentResource(Fleet)
+@added(Versions.v2024_05_02_preview)
+@doc("The AutoUpgradeProfile resource.")
+model AutoUpgradeProfile
+  is Azure.ResourceManager.ProxyResource<AutoUpgradeProfileProperties> {
+  @doc("The name of the AutoUpgradeProfile resource.")
+  @key("autoUpgradeProfileName")
+  @segment("autoUpgradeProfiles")
+  @pattern("^[a-z0-9]([-a-z0-9]*[a-z0-9])?$")
+  @minLength(1)
+  @maxLength(50)
+  @path
+  @visibility("create", "read")
+  name: string;
+
+  ...EntityTagProperty;
```

Comment:
>...EntityTagProperty [](http://example.com/codeflow?start=2&length=20)

Don't add your own top-level tags implementation to a proxy resource type. They won't behave like ARM tags and will be very confusing for clients.

If you want users to be able to tag the resource, consider using a tracked resource. Otherwise, it's not really recommended, but at minimum, the property must be named something other than tags (e.g. labels), and must reside within the property bag.

Reply:
This was typespec warning me that the EntityTag was deprecated, and suggesting to use this instead. We aren't adding anything, see the Swagger not changing.

Reply:
https://azure.github.io/typespec-azure/docs/libraries/azure-resource-manager/reference/data-types#Azure.ResourceManager.EntityTagProperty

Reply:
Can you advise on what we are supposed to do, and why the documented property is not the right approach?
We use etag on all our resources, and we follow the ARM guideline, implemented by this typespec property.

As documented here in Response body section, the etag property is on the envelope:

https://github.com/cloud-and-ai-microsoft/resource-provider-contract/blob/master/v1.0/resource-api-reference.md#response-3

for example, our prior resources, as expected, the eTag property is on the envelope of the resource:

https://github.com/Azure/azure-rest-api-specs/pull/28971/files#diff-2a4c885db09c614d7bbb5a027e6922c8ed5e0b245523972ec1cce58a1ffee32dR2371

Reply:
Re-reading your coment, it seems that you did not understand the purpose of this property?
@mentat9 can you please re-assess asap?

Reply:
@mentat9 @serbrech  `EntityTagProperty`  is for ETags, not for ARM Tags (for which, we expect folks to use TrackedResource)

Reply:
Apologies, that was my mistake: you can ignore that comment.

Reply:
Thanks, is there anything else blocking an approval?